### PR TITLE
refactor: Release to Publish

### PR DESCRIPTION
**Pull request by shift4id**

### CODE REVIEW

**Comment by jnsdls** [comment]
#### 📝 Changed routes:


| [**…Address\]/\[...catchAll\]**](https://thirdweb-ck890ka13.thirdweb-preview.com/[networkOrAddress]/[...catchAll]) | [**/\[networkOrAddress\]**](https://thirdweb-ck890ka13.thirdweb-preview.com/[networkOrAddress]) |
|:-:|:-:|
|<a href="https://thirdweb-ck890ka13.thirdweb-preview.com/[networkOrAddress]/[...catchAll]">Screenshot of /[networkOrAddress]/[...catchAll]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/pYnV0gdr69aMlHgz99Hu5tnQGhumbV-iKEtmqxczZLvvsHdj_SbWWaU6npcWyfFAgCTcNzDQF4QZdmms6ytiBA.png">(view full size)</a> |<a href="https://thirdweb-ck890ka13.thirdweb-preview.com/[networkOrAddress]">Screenshot of /[networkOrAddress]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/GVMzY3cVIAyr-i7IiMxPeGSzI5FRIfc7-RuvAoo6DpWO1Mt0a1j4ev8O68TzzvNz2ClGwhpIxhfe-qHnZSPZuQ.png">(view full size)</a> |



| [**…s/release/\[contractId\]**](https://thirdweb-ck890ka13.thirdweb-preview.com/contracts/release/[contractId]) | [**/contracts/release**](https://thirdweb-ck890ka13.thirdweb-preview.com/contracts/release) |
|:-:|:-:|
|<a href="https://thirdweb-ck890ka13.thirdweb-preview.com/contracts/release/[contractId]">Screenshot of /contracts/release/[contractId]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/zAjC9nXMttm4VVpaZhZhzPYWnBNU0r84S4VMnHFN1nDlCoSdTbimWBkaAmhewDZLqyoY5Iy6rYaM7dt7qdz8XQ.png">(view full size)</a> |<a href="https://thirdweb-ck890ka13.thirdweb-preview.com/contracts/release">Screenshot of /contracts/release</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/dNGRllyy3Sp_MWlzky8bbrjwRV1zfEzUqlkBVkwIcfFnrzCXz9nA6ovrn7JeBG57Lum1XmMlZvXiWakYsAPwnQ.png">(view full size)</a> |



| [**/dashboard**](https://thirdweb-ck890ka13.thirdweb-preview.com/dashboard) | [**/explore**](https://thirdweb-ck890ka13.thirdweb-preview.com/explore) |
|:-:|:-:|
|<a href="https://thirdweb-ck890ka13.thirdweb-preview.com/dashboard">Screenshot of /dashboard</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/WIoAlRgkaFdQO8JzQeB2NZ0wbgWMYGyUJIdIAyQVwdUOiraHHFx7sxpgDLR0GY0NCROyrnvSGziwZ0qupQx12A.png">(view full size)</a> |<a href="https://thirdweb-ck890ka13.thirdweb-preview.com/explore">Screenshot of /explore</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/EOzrEurfakt-X6qPbR3KuHcTrRoGDKH7dHzZxXsRKnPbNYnw1e7umA_8BM0mZAkW5gVcxgmDjJo3TLbhJsqzjQ.png">(view full size)</a> |



And 6 other routes:
- [**/api/og/profile**](https://thirdweb-ck890ka13.thirdweb-preview.com/api/og/profile)
- [**/api/og/publish**](https://thirdweb-ck890ka13.thirdweb-preview.com/api/og/publish)
- [**/api/revalidate/publish**](https://thirdweb-ck890ka13.thirdweb-preview.com/api/revalidate/publish)
- [**/api/verify**](https://thirdweb-ck890ka13.thirdweb-preview.com/api/verify)
- [**/programs**](https://thirdweb-ck890ka13.thirdweb-preview.com/programs)
- [**/publish**](https://thirdweb-ck890ka13.thirdweb-preview.com/publish)



Commit f7897525427c762d77923d5e3e7ab885b252a90a (https://thirdweb-ck890ka13.thirdweb-preview.com).

File: components/contract-components/contract-publish-form/factory-fieldset.tsx
Comment (nachoiacovino) [inline components/contract-components/contract-publish-form/factory-fieldset.tsx:33]: we cannot change this if we don't change it on portal, who's handling that?

File: components/contract-components/contract-publish-form/landing-fieldset.tsx
Comment (nachoiacovino) [inline components/contract-components/contract-publish-form/landing-fieldset.tsx:324]: same here (and all portal links)

File: components/contract-components/hooks.ts
Comment (nachoiacovino) [inline components/contract-components/hooks.ts:139]: is this not necessary?

File: components/contract-components/published-contract/index.tsx
Comment (nachoiacovino) [inline components/contract-components/published-contract/index.tsx:135]: I guess we can also change it here to be `publishDate`? not super important

File: page-id.ts
Comment (nachoiacovino) [inline page-id.ts:19]: I'm not sure I would change the pageId since they would break posthog history but I defer to @jnsdls on this one

File: redirects.js
Comment (nachoiacovino) [inline redirects.js:33]: we need a redirect here for `/release` to `/publish` for the landing

File: pages/publish.tsx
Comment (nachoiacovino) [inline pages/publish.tsx:39]: We need a new image here with the new CLI command.

File: redirects.js
Comment (nachoiacovino) [reply redirects.js:33]: Also another one for any match, so for example:

https://thirdweb-www-git-sid-release-to-publish.thirdweb-preview.com/contracts/release/QmY3ZMRSQUN3WGhdqt7jCdviA6Jj16b8iN1C954pZSDpGX%2F0

would bring me to the correct url

we can do it by adding this:

```
{
  source: "/contracts/release/:path*",
  destination: "/contracts/publish/:path*",
  permanent: false,
},
```

File: pages/publish.tsx
Comment (nachoiacovino) [inline pages/publish.tsx:84]: Can we just a <Text> here instead of <p>?

Very minor stylistic choice.

**Review by nachoiacovino — DISMISSED** [review]
We also need a new hero image for publish

image

Hero image and og image are not blockers, but we cannot ship without setting the redirects and linking to 404s on portal, so these need to be addressed before merging.

Regarding portal, feel free to sync with Jarrod or go and fix it yourself, besides changing the wording everywhere, redirects also would need to be set on portal.

**Comment by nachoiacovino** [comment]
@shift4id can you fix the merge conflicts?

**Pull request merged**
